### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by JaredBorders**
Smart Margin v2.0.0

https://kips.kwenta.io/kips/kip-55/

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,16 @@
-node_modules
```
Comment (tommyrharper) [inline .gitignore:1]: shouldn't we keep `node_modules` here?

File: .gitignore [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+# Dotenv file
 .env
-coverage
-coverage.json
-typechain
-.DS_store
 
-#Hardhat files
-cache
-artifacts
```
Comment (tommyrharper) [inline .gitignore:10]: Also artifacts as well

File: .gitignore (hunk 1 above)
Comment (JaredBorders) [reply .gitignore:1]: There are no modules. https://github.com/Kwenta/margin-manager/blob/v2.0.0/package.json

File: .gitignore (hunk 2 above)
Comment (JaredBorders) [reply .gitignore:10]: There are no artifacts. The project no longer uses hardhat

**Comment by tommyrharper** [comment]
Github won't let me comment directly because the file is too long, but in `test/unit/Account.t.sol` you have the following command shortcut:
```solidity
    function withdrawEth(uint256 amount) private {
        IAccount.Command[] memory commands = new IAccount.Command[](1);
        commands[0] = IAccount.Command.ACCOUNT_MODIFY_MARGIN;
        bytes[] memory inputs = new bytes[](1);
        inputs[0] = abi.encode(amount);
        account.execute(commands, inputs);
    }
```

I think `line 685` should be modified from `commands[0] = IAccount.Command.ACCOUNT_MODIFY_MARGIN;` to `commands[0] = IAccount.Command.ACCOUNT_WITHDRAW_ETH;`

File: src/interfaces/IEvents.sol [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice returns the address of the factory contract
+    function factory() external view returns (address);
+
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice emitted after a successful withdrawal
```
Comment (tommyrharper) [inline src/interfaces/IEvents.sol:27]:
```suggestion
    /// @notice emitted after a successful deposit
```

**Pull request merged**